src/literal.h

```diff
@@ -33,25 +33,27 @@ class Literals;
 struct ExceptionPackage;
 
 class Literal {
+
+private:
   // store only integers, whose bits are deterministic. floats
   // can have their signalling bit set, for example.
   union {
     int32_t i32;
     int64_t i64;
     uint8_t v128[16];
-    Name func; // function name for funcref
+    // funcref function name. not set indicates `null`.
+    Name func;
+    // exnref package. not set indicates `null`.
     std::unique_ptr<ExceptionPackage> exn;
+    // literals of other reference types can only be `null`
```

Comment:
This seems like a big limitation. Are you planning to allow non-null literals for these types later? I would expect that the interpreter, at least, would depend on being able to have arbitrary literals for all types.

Reply:
This is mostly a "for now", since knowing that an `externref` is `null` can already be useful during optimizations (i.e. eliminating a `ref.is_null` or a local if it's known to be assigned a `null`), but there is no good story yet on representing arbitrary external values as a literal. Would expect that being able to recognize extern values doesn't give us much just yet, but there might be (future) use cases I don't see yet, for example someone using the interpreter standalone? If that'd ever happen, I'd expect these literals to be represented by an `ExternValue` or similar `struct`, similar to what's currently done for `ExceptionPackage`.

Reply:
Thinking a little more about this, there isn't really a functional change here as well. Previously, we had `Literal(nullref)` representing `null`s, so the only change is that instead of encoding this information in the type, we now encode it along the type, for each ref type, and instead of checking `literal.type == nullref` we check `literal.isNull()`. There wasn't a story for `externref` of non-null values before, and there still isn't one now.

Reply:
I think we will need this capability to 1) run the spec tests and fuzzer with reference types enabled and 2) avoid bailing out the first time we see a reference typed value in Precompute. No need to address this in this PR, of course, but it would be good to have a TODO.